### PR TITLE
Fix error in documentation

### LINKED ISSUE #17438: Documentation contains an error
````
import akka.pattern.ask

class ExampleActor extends Actor {
  val other = context.actorOf(Props[OtherActor], "childName") // will be destroyed and re-created upon restart by default

  def receive {
    case Request1(msg) => other ! refine(msg)     // uses this actor as sender reference, reply goes to us
    case Request2(msg) => other.tell(msg, sender()) // forward sender reference, enabling direct reply
    case Request3(msg) =>
      implicit val timeout = Timeout(5.seconds)
      sender() ! (other ? msg)  // will reply with a Future for holding other's reply
  }
}
````

should be 

```
 * import akka.pattern.ask
 * import scala.concurrent.Await
 *
 * class ExampleActor extends Actor {
 *   val other = context.actorOf(Props[OtherActor], "childName") // will be destroyed and re-created upon restart by default
 *
 *   def receive {
 *     case Request1(msg) => other ! refine(msg)     // uses this actor as sender reference, reply goes to us
 *     case Request2(msg) => other.tell(msg, sender()) // forward sender reference, enabling direct reply
 *     case Request3(msg) =>
 *       implicit val timeout = Timeout(5.seconds)
 *       sender() ! Await.result(other ? msg, 2 * timeout)  // will reply with a Future for holding other's reply
 *   }
 * }
```

### PR DESCRIPTION
**Pull request by gkb**
It doesn't make sense to send a message composed of a future to another
actor. Futures aren't serializable.

Fixes https://github.com/akka/akka/issues/17438

### CODE REVIEW

File: akka-actor/src/main/scala/akka/actor/ActorRef.scala [hunk 1]
```diff
@@ -48,7 +49,7 @@ object ActorRef {
  *     case Request2(msg) => other.tell(msg, sender()) // forward sender reference, enabling direct reply
  *     case Request3(msg) =>
  *       implicit val timeout = Timeout(5.seconds)
- *       sender() ! (other ? msg)  // will reply with a Future for holding other's reply
+ *       (other ? msg) pipeTo sender() // will reply with a Future for holding other's reply
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/actor/ActorRef.scala]: comment should be updated too.

Comment (gkb) [inline akka-actor/src/main/scala/akka/actor/ActorRef.scala]: Done

File: akka-actor/src/main/scala/akka/actor/ActorRef.scala [hunk 2]
```diff
@@ -48,7 +49,7 @@ object ActorRef {
  *     case Request2(msg) => other.tell(msg, sender()) // forward sender reference, enabling direct reply
  *     case Request3(msg) =>
  *       implicit val timeout = Timeout(5.seconds)
- *       sender() ! (other ? msg)  // will reply with a Future for holding other's reply
+ *       (other ? msg) pipeTo sender() // will pipe the Future from other back to sender
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/actor/ActorRef.scala]: pipe the value of the future not the future it's self

**Comment by hepin1989** [comment]
and here:
https://github.com/akka/akka/blob/master/CONTRIBUTING.md

after you finished you should rebase and squash the commit into one.
like ```!doc #{issue number} {commit message}```

**Comment by gkb** [comment]
Done.

**Comment by patriknw** [comment]
OK TO TEST

**Comment by patriknw** [comment]
LGTM, thanks

File: akka-actor/src/main/scala/akka/actor/ActorRef.scala [hunk 3]
```diff
@@ -48,7 +49,7 @@ object ActorRef {
  *     case Request2(msg) => other.tell(msg, sender()) // forward sender reference, enabling direct reply
  *     case Request3(msg) =>
  *       implicit val timeout = Timeout(5.seconds)
- *       sender() ! (other ? msg)  // will reply with a Future for holding other's reply
+ *       (other ? msg) pipeTo sender() // will get a Future from other and pipe its value to sender
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/actor/ActorRef.scala]: ```from other``` I don;t think so.
the ```ask``` return a ```future``` of the other's reply.and the future is comes from the local ask support actor not the other that is asked.
how about 
```the ask call will get a Future of other's first reply,and then we pipe its value to the sender when it complete.```

Comment (gkb) [inline akka-actor/src/main/scala/akka/actor/ActorRef.scala]: Fixed it.

**Comment by drewhk** [comment]
Does this doc example have a Java counterpart? If yes, then it should be updated, too, and then we can merge this.

**Comment by gkb** [comment]
I updated the Java counterpart as well.

**Comment by gkb** [comment]
Is there anything else this pull request needs before it gets merged?

**Comment by ktoso** [comment]
PLS BUILD

**Comment by ktoso** [comment]
@gkb Thanks for the contrib! It's a nice one. We just need to run PR validation once more, since now our build is using JDK8 and the javadoc generation is a bit stricter. Once it passes validation we'll merge.

**Pull request merged**